tests/framework/extensions/machinepools/machinepools.go

```diff
@@ -101,3 +162,74 @@ func CreateAllMachinePools(nodeRoles []NodeRoles, machineConfig *v1.SteveAPIObje
 
 	return machinePools
 }
+
+func CreateNewWorkerMachinePool(client *rancher.Client, cluster *v1.SteveAPIObject, updatedCluster *apisV1.Cluster, machineConfig *v1.SteveAPIObject, quantity int32) (error, *v1.SteveAPIObject, *apisV1.Cluster) {
+	nodePoolConfig := []NodeRoles{
+		{
+			ControlPlane: false,
+			Etcd:         false,
+			Worker:       true,
+			Quantity:     1,
+		},
+	}
+
+	machinePools := CreateAllMachinePools(nodePoolConfig, machineConfig, nil)
+	machinePools[0].Name = "auto-pool-" + namegenerator.RandStringLower(5)
+	updatedCluster.Spec.RKEConfig.MachinePools = append(updatedCluster.Spec.RKEConfig.MachinePools, machinePools...)
+
+	adminClient, err := rancher.NewClient(client.RancherConfig.AdminToken, client.Session)
+	if err != nil {
+		return err, nil, nil
+	}
+
+	logrus.Infof("Creating new worker machine pool...")
+
+	updatedClusterResp, err := updateClusterMachinePool(adminClient, cluster, updatedCluster, quantity)
+	if err != nil {
+		return err, nil, nil
+	}
+
+	return nil, updatedClusterResp, updatedCluster
+}
+
+func ScaleNewWorkerMachinePool(client *rancher.Client, updatedClusterResp *v1.SteveAPIObject, updatedCluster *apisV1.Cluster, quantity int32) (error, *v1.SteveAPIObject) {
```

Comment:
one more suggestion; I'm not sure how this would work exactly, but when I read 'scale' in the function name, I sort of expect the number provided to the function to scale by that number, not to be the new number that is scaled to. Would it make sense to add/subtract the number requested to the specified pool? 

i.e. right now, if my pool is at 3 and I want 4 you send 4 as the quantity argument. 

Why not send `1`, and update the quantity for the user by 1? similarly if we want to scale down, send `-1`. This removes the obligation from the user to know how many nodes are in the pool

Reply:
After offline discussion, addressed this concern to scale up and down by one. Marking as resolved.